src/types.jl

```diff
@@ -38,9 +38,10 @@ end
 mutable struct File <: H5DataStore
     id::API.hid_t
     filename::String
+    track_order::Bool
 
-    function File(id, filename, toclose::Bool=true)
-        f = new(id, filename)
+    function File(id, filename, fcpl::FileCreateProperties=FileCreateProperties(), toclose::Bool=true)
```

Comment:
Since we are getting a file id as an argument, shouldn't we get the `FileCreateProperty` from the file id via `H5Fget_create_plist`?

https://docs.hdfgroup.org/hdf5/develop/group___h5_f.html#ga2f823a9e929b00b06a6be80619a61778

Also why do we need to add a `track_order` property to the File struct?  Should it basically be a property of the root group?

Reply:
See the comments, below this commit was obsolete, I've updated the PR.